front_end/models/react_native/ReactDevToolsBindingsModel.ts

```diff
@@ -115,23 +110,117 @@ export class ReactDevToolsBindingsModel extends SDK.SDKModel.SDKModel {
   }
 
-  async enable(target: SDK.Target.Target): Promise<void> {
-    const runtimeModel = target.model(SDK.RuntimeModel.RuntimeModel);
+  async enable(): Promise<void> {
+    if (this.enabled) {
+      throw new Error('ReactDevToolsBindingsModel is already enabled');
+    }
+
+    const runtimeModel = this.target().model(SDK.RuntimeModel.RuntimeModel);
+    if (!runtimeModel) {
+      throw new Error('Failed to enable ReactDevToolsBindingsModel: runtime model is not available');
+    }
+
+    await this.waitForFuseboxDispatcherToBeInitialized()
+      .then(() => runtimeModel.agent.invoke_evaluate({expression: `${RUNTIME_GLOBAL}.BINDING_NAME`}))
+      .then(response => {
+        if (response.exceptionDetails) {
+          throw new Error('Failed to get binding name for ReactDevToolsBindingsModel on a global: ' + response.exceptionDetails.text);
+        }
+
+        if (response.result.value === null || response.result.value === undefined) {
+          throw new Error('Failed to get binding name for ReactDevToolsBindingsModel on a global: returned value is ' + String(response.result.value));
+        }
+
+        if (response.result.value === '') {
+          throw new Error('Failed to get binding name for ReactDevToolsBindingsModel on a global: returned value is an empty string');
+        }
+
+        return response.result.value;
+      })
+      .then(bindingName => {
+        this.messagingBindingName = bindingName;
+        runtimeModel.addEventListener(SDK.RuntimeModel.Events.BindingCalled, this.bindingCalled, this);
+
+        return runtimeModel.agent.invoke_addBinding({name: bindingName});
+      })
+      .then(response => {
+        const possiblyError = response.getError();
+        if (possiblyError) {
+          throw new Error('Failed to add binding for ReactDevToolsBindingsModel: ' + possiblyError);
+        }
+
+        this.enabled = true;
+        this.initializeExecutionContextListeners();
+      });
+  }
+
+  isEnabled(): boolean {
+    return this.enabled;
+  }
+
+  private initializeExecutionContextListeners(): void {
+    const runtimeModel = this.target().model(SDK.RuntimeModel.RuntimeModel);
+    if (!runtimeModel) {
+      throw new Error('Failed to initialize execution context listeners for ReactDevToolsBindingsModel: runtime model is not available');
+    }
+
+    runtimeModel.addEventListener(SDK.RuntimeModel.Events.ExecutionContextCreated, this.onExecutionContextCreated, this);
+    runtimeModel.addEventListener(SDK.RuntimeModel.Events.ExecutionContextDestroyed, this.onExecutionContextDestroyed, this);
+  }
+
+  private onExecutionContextCreated({data: executionContext}: ContextCreatedEvent): void {
+    if (executionContext.name !== MAIN_EXECUTION_CONTEXT_NAME) {
+      return;
+    }
+
+    void this.waitForFuseboxDispatcherToBeInitialized()
+      .then(() => this.dispatchEventToListeners(Events.BackendExecutionContextCreated));
```

Comment:
Just a reminder to myself, I should probably add `catch` block here and send some event, which will be picked up by `ReactDevToolsModel`, and then `ReactDevToolsView` to show an error view.